lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -4564,6 +4573,116 @@ void glow::convertQuantizedConstants(Function *F, CompilationContext &cctx) {
+    bool isSupported = B.isOpSupported(node);
+
+    // check if all of the outputs of node are quantized type
+    bool isAllResQuantized = true;
+    for (size_t i = 0, e = node.getNumResults(); i < e; i++) {
+      if (node.getType(i)->isFPType()) {
+        isAllResQuantized = false;
+        break;
+      }
+    }
+
+    if (!isSupported && isAllResQuantized) {
+      // Check for FP16 support of the node
+      if (precConfig.convertToFP16) {
+        std::vector<TypeRef> inputTypes, outputTypes;
+        TypeRef inpType = nullptr, outType = nullptr;
+
+        for (size_t i = 0, e = node.getNumResults(); i < e; i++) {
+          if (node.getType(i)->isQuantizedType()) {
+            outType =
+                F->getParent()->uniqueType(ElemKind::Float16Ty, node.dims(i));
+          } else {
+            outType = node.getType(i);
+          }
+          outputTypes.push_back(outType);
+        }
+
+        for (size_t i = 0, e = node.getNumInputs(); i < e; i++) {
+          if (node.getNthInput(i).getType()->isQuantizedType()) {
+            inpType = F->getParent()->uniqueType(ElemKind::Float16Ty,
+                                                 node.getNthInput(i).dims());
+          } else {
+            inpType = node.getNthInput(i).getType();
+          }
+          inputTypes.push_back(inpType);
+        }
+
+        NodeInfo fp16node = NodeInfo(node.getKind(), inputTypes, outputTypes);
+        isFP16Supported = B.isOpSupported(fp16node);
+      }
+
+      // Create a Cloned Node
+      Node *clonedNode = node.clone();
+      TypeRef newTy = nullptr;
+      for (size_t i = 0, e = node.getNumResults(); i < e; i++) {
+        if (node.getType(i)->isQuantizedType()) {
+          if (isFP16Supported) {
+            newTy =
+                F->getParent()->uniqueType(ElemKind::Float16Ty, node.dims(i));
+          } else {
+            newTy = F->getParent()->uniqueType(ElemKind::FloatTy, node.dims(i));
+          }
+        } else {
+          newTy = node.getType(i);
+        }
+        clonedNode->setType(i, newTy);
+      }
+      F->addNode(clonedNode);
+
+      // Add input dequantize node
+      for (size_t i = 0, e = node.getNumInputs(); i < e; i++) {
+        auto input = node.getNthInput(i);
+        TypeRef inTy = nullptr;
+        if (input.getType()->isQuantizedType()) {
+          if (isFP16Supported) {
+            inTy = F->getParent()->uniqueType(ElemKind::Float16Ty,
+                                              input.getType()->dims());
+          } else {
+            inTy = F->getParent()->uniqueType(ElemKind::FloatTy,
+                                              input.getType()->dims());
+          }
+          std::string deqName =
+              node.getName().str() + "_dequantize" + std::to_string(i);
+          auto *DQ = F->createDequantize(deqName, input, inTy);
+          clonedNode->setNthInput(i, DQ);
+        } else {
+          clonedNode->setNthInput(i, input);
+        }
```

Comment:
Similar to above to simplify.
```suggestion
        if (!input.getType()->isQuantizedType()) {
          continue;
        }
        auto input = node.getNthInput(i);
        ElemKind k = isFP16Supported ? ElemKind::Float16Ty : ElemKind::FloatTy;
        TypeRef inTy = F->getParent()->uniqueType(k, input.dims());
        const std::string deqName =
            node.getName().str() + "_dequantize" + std::to_string(i);
        auto *DQ = F->createDequantize(deqName, input, inTy);
        clonedNode->setNthInput(i, DQ);
```

Reply:
Will do.